### PR TITLE
Add a 404 page to display when a file has been deleted

### PR DESCRIPTION
**Pull request by franky-fk**
### Issue
 - When a publicly shared file is deleted in [Explorer](https://beta-broker.opacitynodes.com:3000/admin/delete), the public URL still shows the file details but not the file itself since it no longer exists.
### Solution
 - When a public URL of a deleted file is visited, a 404 page should be displayed instead of the details of a file that no longer exists.
### How to test locally
 - Clone the repository and and cd into `/front`
 - Install dependencies by running `npm install`
 - Start the project by running `npm start`
 - A deleted file will always have a file size of 0. Therefore, got to `src/shortlink_dev.html` and under the script tag where there is `window.OpacityFile`, change the value of `fileSizeBytes` to '0'.
 - You should now view a 404 page being displayed
### Screenshot
Screenshot 2022-05-11 at 8 24 02 PM

### CODE REVIEW

File: front/src/pages/SharePage/SharePage.tsx [hunk 1]
```diff
@@ -28,6 +29,14 @@ declare global {
...
   const file = useMemo(() => window.OpacityFile, [window.OpacityFile])
   const [pageLoading, setPageLoading] = useState(false)
+  const [error, setError] = useState(false)
+
+
+  useEffect(() => {
+    if (file.fileSizeBytes == "0") {
+      setError(true)
+    }
+  })
```
Comment (faizan-firekamp) [inline front/src/pages/SharePage/SharePage.tsx:39]: lets not even use state, we can just check `file.fileSizeBytes` directly

**Review by faizan-firekamp — CHANGES_REQUESTED** [review]
Please make the change and test

File: front/src/pages/SharePage/SharePage.tsx (hunk 1 above)
Comment (franky-fk) [reply front/src/pages/SharePage/SharePage.tsx:39]: @faizan-firekamp I tried that and it was throwing an error. Something about re-rendering several times.

**Review by faizan-firekamp — APPROVED** [review]
Looks good

File: front/src/pages/SharePage/SharePage.tsx [hunk 2]
```diff
@@ -3,80 +3,81 @@ import SiteWrapper from "../../SiteWrapper";
...
-  }
+      .then((res) => res.blob())
+      .then((res) => {
+        const blob = new Blob([res]);
+        saveAs(blob, file.title);
+        setPageLoading(false);
+      });
+  };
 
+  if (Number(file.fileSizeBytes) === 0) {
```
Comment (kamranpirwani) [inline front/src/pages/SharePage/SharePage.tsx:44]: For the file handle I see that's deleted, this should also be catching on the fetch in L35 in order to redirect to 404

**Pull request merged**